### DISCUSSION TITLE
Is it possible to import a macOS Photos Library? (El Capitan)

### DISCUSSION THREAD

**Discussion by TheTechRobo** [photoprism/photoprism#2078] [Q&A] [answered]
My parents are using an old iMac on El Capitan/the Photos app for their photo management. It's really slow and clunky. I'd like to migrate them to a program like this; is there a simple way to import such a library? Thanks!

**Comment by graciousgrey** [comment, marked as answer]
This https://docs.photoprism.app/user-guide/use-cases/apple/ has been tested with the Photos App on Catalina and BigSur. Might work for El Capitan as well.

In general it's important to export the original files from the Photos App ideally together with metadata files such as xmp. Then you move  all files/folders to PhotoPrism's import or originals folder, depending on what method you want to use (https://docs.photoprism.app/user-guide/library/).

**Reply by tlvince** [reply]
With regards to the export recommendation, has importing directly from the `originals` directory (within `Photos Library.photoslibrary/originals`) been tested? Do we loose anything other than the original filename and the additional [metadata](https://docs.photoprism.app/user-guide/use-cases/apple/#metadata) set by Photos?

**Reply by graciousgrey** [reply]
You mean configuring the folder Apple Photos uses to store media as PhotoPrism's import directory?

**Reply by tlvince** [reply]
I mean configuring PhotoPrism to import directly from Apple Photos' `originals` directory rather than exporting from Apple Photos.

**Reply by graciousgrey** [reply]
We did not test this ourselves. But in general you can configure whatever directory you want (if it has the right permissions) as PhotoPrism's originals or import directory. 

In case Apple does not store xmp files alongside the original files, PhotoPrism cannot read metadata from it. So metadata you added using Apple Photos will be lost.

Also when using import instead of index, the files will be renamed and sorted by year/month (https://docs.photoprism.app/user-guide/library/) so you will have a different folder structure afterwards. The original name will be stored as file property. 
If you are unsure, you can try it with a fresh PhotoPrism install and the import+copy option. This way files from Apples originals directory are copied (not moved) to PhotoPrism's originals directory.

**Reply by tlvince** [reply]
Thanks. I tested it and all's fine 👍🏻
